### ISSUE TITLE
Auto generate titles

### ISSUE THREAD

**Issue by ang-zeyu** [MarkBind/markbind#2297]
### Please confirm that you have searched existing issues in the repo

Yes, I have searched the existing issues

### Any related issues?

#2166

### Is your feature request related to a problem? Please describe.

https://github.com/MarkBind/markbind/issues/2166#issuecomment-1464909988

### Describe the solution you'd like

We can consider auto generating page titles using file names

**Comment by lhw-1** [comment]
As I mentioned in the linked comment, if we were to implement an auto-generated title, the best way to do this would be to extract the file name, capitalize it (and perhaps some further processing like removing underscores `_`), and use it as the page title. While MKDocs and Jekyll have played around with this idea, there doesn't seem to be a good way to make sure that a proper title is extracted in all scenarios, though we can probably start small (e.g. implementing just the capitalization).

Note that #2166 discussed this in the context of empty titles (i.e. no titles are provided in either the `frontmatter` or `site.json`), but this issue can be expanded to cover other scenarios. Of course, this would require our title auto-generation to be robust.

At the moment we do not have a pressing need for auto-generated title, but it may be very useful for those who want to convert a large set of documents into MarkBind & define a new sensible title for each of the pages. 

(Maybe we can even consider a CLI command flag for `markbind init --convert` that can take in a regex for title generation? Just a thought)

**Comment by tlylt** [comment]
> As I mentioned in the linked comment, if we were to implement an auto-generated title, the best way to do this would be to extract the file name, capitalize it (and perhaps some further processing like removing underscores `_`), and use it as the page title. While MKDocs and Jekyll have played around with this idea, there doesn't seem to be a good way to make sure that a proper title is extracted in all scenarios, though we can probably start small (e.g. implementing just the capitalization).

My personal experience would be that a simple relationship of file name -> page title doesn't work well for blog articles as the name of an article can be quite long. However, not everyone likes file names to be long.

Creating a page title containing proper capitalization and special characters and then translating that into a file name just so that it is auto-converted back into the intended form may not be an ideal experience.

> At the moment we do not have a pressing need for auto-generated title, but it may be very useful for those who want to convert a large set of documents into MarkBind & define a new sensible title for each of the pages.

I agree, definitely useful for the mass conversion case. For example, if a site has a number of files each containing a coding question. Then, converting `question/1.md` and `question/2.md` into `Question 1` & `Question 2` (or some other variants) might be helpful.

**Comment by lhw-1** [comment]
> My personal experience would be that a simple relationship of file name -> page title doesn't work well for blog articles as the name of an article can be quite long. However, not everyone likes file names to be long.
> 
> Creating a page title containing proper capitalization and special characters and then translating that into a file name just so that it is auto-converted back into the intended form may not be an ideal experience.

Right, we'll need to ensure that if this feature was to be implemented, it would have to be as an optional feature.

> I agree, definitely useful for the mass conversion case. For example, if a site has a number of files each containing a coding question. Then, converting `question/1.md` and `question/2.md` into `Question 1` & `Question 2` (or some other variants) might be helpful.

I would imagine that you can specify a flag like `--auto-title` and maybe specify a directory / directories for title auto-generation, like: `markbind init --convert --auto-title docs/questions/ docs/answers/`, which will automatically generate a title for all the files under the specified directories.

This would be a feature intended for mass-conversion, and to allow flexibility, we can consider extending this feature in the future to allow pre-defined regex / title formats.
